hub-saml/src/main/java/uk/gov/ida/saml/hub/validators/response/idp/IdpFailedSamlAssertionsDecryption.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.ida.saml.hub.validators.response.idp;
+
+import io.prometheus.client.Counter;
+import org.slf4j.event.Level;
+import uk.gov.ida.saml.core.security.FailedSamlAssertionsDecryption;
+import uk.gov.ida.saml.security.exception.SamlFailedToDecryptException;
+import uk.gov.ida.saml.security.validators.ValidatedResponse;
+
+public class IdpFailedSamlAssertionsDecryption implements FailedSamlAssertionsDecryption {
+    
+    private static final Counter idpDecryptionErrorCounter = Counter.build(
```

Comment:
I keep thinking it would be nice to pass this in, rather than hard code it like this. But I know it's awkward to do...

Reply:
It was really awkward, lots of changes and it broke the integration test, so I still have this as static counters. I have lifted a static method that does some reflection thingy that allows for this to be tested.